src/org/zaproxy/zap/extension/websocket/manualsend/WebSocketPanelSender.java

```diff
@@ -80,14 +86,59 @@ public void cleanup() {
     }
     
     private WebSocketProxy getDelegate(Integer channelId) {
-    	if (!connectedProxies.containsKey(channelId)) {
+    	if (!connectedProxies.containsKey(channelId) && !closedProxies.containsKey(channelId)) {
     		logger.warn("Selected WebSocket channel is not connected. Unable to send manual crafted message!");
     		throw new IllegalArgumentException(Constant.messages.getString("websocket.manual_send.fail.disconnected_channel") 
     				+ " " + Constant.messages.getString("websocket.manual_send.fail"));
-    	}
+    	}else if (closedProxies.containsKey(channelId)){
+			logger.warn("Selected WebSocket channel is not connected. Unable to send manual crafted message!");
+			throw new IllegalArgumentException(Constant.messages.getString("websocket.manual_send.fail.disconnected_channel")
+					+ " " + Constant.messages.getString("websocket.manual_send.use_reopen"));
+		}
         return connectedProxies.get(channelId);
     }
 
+    /**
+     * Reopen a new WebSocket Connection same as the channel it gets. Make use of {@link WebSocketReopenSend} to establish the new connection.
+     * @param channelId The connection to re-open
+     * @param newSecWebSocketKey New Sec-WebSocket-Key, null to use the same key as the initial connection
+     * @param useTrackingSessionState useTrackingSessionState false for stateless http request
+     * @param followRedirects true to follow redirections
+     * @return Returns false if fails to re-open the connection
+     */
+    public boolean reOpenChannel(Integer channelId, String newSecWebSocketKey, boolean useTrackingSessionState, boolean followRedirects) {
+        if(webSocketReopenSend == null){
+            webSocketReopenSend = new WebSocketReopenSend();
+            webSocketReopenSend.addPersistentConnectionListener(Control.getSingleton().getExtensionLoader().getExtension(ExtensionWebSocket.class));
+        }
+        return webSocketReopenSend.reEstablishConnection(getWebSocketProxyFromChannelID(channelId), newSecWebSocketKey, useTrackingSessionState, followRedirects);
+    }
+
+	public String getSecWebKeyFrom(Integer channelId) throws DatabaseException, HttpMalformedHeaderException {
+		return getWebSocketProxyFromChannelID(channelId).getHandshakeReference().getHttpMessage().getRequestHeader().getHeader("Sec-WebSocket-Key");

```

Comment:
May as well use the public constant `WebSocketReopenSend.HTTP_HEADER_SEC_WEBSOCKET_KEY`